File: customer-mitra/frontend/lib/models/user_role.dart
/// Enum untuk role user di aplikasi Nebeng
enum UserRole {
  customer('customer'),
  mitra('mitra'),
  posmitra('posmitra'),
  admin('admin'),
  superadmin('superadmin'),
  finance('finance');

  final String value;

  const UserRole(this.value);

  /// Convert dari string ke UserRole enum
  static UserRole fromString(String role) {
    switch (role.toLowerCase()) {
      case 'customer':
        return UserRole.customer;
      case 'mitra':
        return UserRole.mitra;
      case 'posmitra':
      case 'pos_mitra':
        return UserRole.posmitra;
      case 'admin':
        return UserRole.admin;
      case 'superadmin':
      case 'super_admin':
        return UserRole.superadmin;
      case 'finance':
        return UserRole.finance;
      default:
        return UserRole.customer; // default fallback
    }
  }

  /// Check apakah role adalah customer
  bool get isCustomer => this == UserRole.customer;

  /// Check apakah role adalah mitra
  bool get isMitra => this == UserRole.mitra;

  /// Check apakah role adalah pos mitra
  bool get isPosMitra => this == UserRole.posmitra;

  /// Check apakah role adalah admin
  bool get isAdmin => this == UserRole.admin;

  /// Check apakah role adalah superadmin
  bool get isSuperAdmin => this == UserRole.superadmin;

  /// Check apakah role adalah finance
  bool get isFinance => this == UserRole.finance;

  @override
  String toString() => value;
}

File: backend/app/Enums/UserRole.php
<?php

namespace App\Enums;

enum UserRole: string
{
    case CUSTOMER = 'customer';
    case MITRA = 'mitra';
    case POSMITRA = 'posmitra';
    case FINANCE = 'finance';
    case ADMIN = 'admin';
    case SUPERADMIN = 'superadmin';

    /**
     * Get all role values as array
     */
    public static function values(): array
    {
        return array_column(self::cases(), 'value');
    }

    /**
     * Check if the role is customer
     */
    public function isCustomer(): bool
    {
        return $this === self::CUSTOMER;
    }

    /**
     * Check if the role is mitra
     */
    public function isMitra(): bool
    {
        return $this === self::MITRA;
    }

    /**
     * Check if the role is pos mitra
     */
    public function isPosMitra(): bool
    {
        return $this === self::POSMITRA;
    }

    /**
     * Check if the role is finance
     */
    public function isFinance(): bool
    {
        return $this === self::FINANCE;
    }

    /**
     * Check if the role is admin
     */
    public function isAdmin(): bool
    {
        return $this === self::ADMIN;
    }

    /**
     * Check if the role is superadmin
     */
    public function isSuperAdmin(): bool
    {
        return $this === self::SUPERADMIN;
    }

    /**
     * Get label for display
     */
    public function label(): string
    {
        return match($this) {
            self::CUSTOMER => 'Customer',
            self::MITRA => 'Mitra',
            self::POSMITRA => 'Pos Mitra',
            self::FINANCE => 'Finance',
            self::ADMIN => 'Admin',
            self::SUPERADMIN => 'Super Admin',
        };
    }
}
